tests/unit_test/channels/whatsapp_test.py

```
@@ -16,6 +17,214 @@
+                args, kwargs = mock_message.call_args
+
+                assert args[0] == bot
+                user_message = args[1]
+
+                assert user_message.text == 'hi'
+
+    @pytest.mark.asyncio
+    async def test_whatsapp_valid_button_message_request(self):
+        from kairon.chat.handlers.channels.whatsapp import Whatsapp, WhatsappBot
+        with patch.object(WhatsappBot, "mark_as_read"):
+            with patch.object(Whatsapp, "process_message") as mock_message:
```

Ruff

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)